src/hpp/gepetto/viewer_factory.py

```diff
@@ -99,13 +107,23 @@ def __call__ (self, args):
         l = locals ();
         self.guiRequest.append ((Viewer.__call__, l));
 
+
+
     ## Create a client to \c gepetto-viewer-server and send stored commands
     #
     # The arguments of Viewer.__init__ can be passed through kwargs
-    def createViewer (self, ViewerClass = Viewer, viewerClient = None, host = None, *args, **kwargs):
-        if host is not None and viewerClient is None:
+    def createViewer (self, ViewerClass = Viewer, viewerClient = None, ghost = False, host = None, *args, **kwargs):
+        if (host is not None or ghost) and viewerClient is None:
             from gepetto.corbaserver import Client as GuiClient
-            viewerClient = GuiClient (host = host)
```

Comment:
I think you can call directly the function `gepetto.corbaserver.gui_client(host=host, ghost=ghost, dont_raise=ghost)`

Reply:
This will behave as follows.
If GV is accessible, then connect to it. If it is not, then,
- if ghost is true, then return a ghost client,
- otherwise, raise an exception.

Reply:
If I understand correctly, the method gui_client only return the gui object. Here I need a client which have the gui object as member, right ? Thats the point of the class I created above.

Reply:
You are right.

In the long term, class GuiClient should be deleted so we should start modifying user code not to rely on it. But this is out of the scope of this PR so you can forget about my comment.